openssh/config.sls

```diff
@@ -3,17 +3,35 @@
 include:
   - openssh
 
-{% if salt['pillar.get']('sshd_config', False) %}
-sshd_config:
+{%- macro sshd_config_file(name) %}
   file.managed:
-    - name: {{ openssh.sshd_config }}
+    - name: {{ name }}
     - source: {{ openssh.sshd_config_src }}
     - template: jinja
     - user: {{ openssh.sshd_config_user }}
     - group: {{ openssh.sshd_config_group }}
     - mode: {{ openssh.sshd_config_mode }}
     - watch_in:
       - service: {{ openssh.service }}
+{%- endmacro %}
+
+{% if salt['pillar.get']('sshd_config', False) %}
+{% set test_config = openssh.sshd_config+'.saltstack-test' %}
+
+sshd_config_test_file:
+{{ sshd_config_file(test_config) }}
+
+{# Test the config before applying it to prevent lock-out. #}
+ssh_config_test:
+  cmd.run:
+    - name: "sshd -t -f {{ test_config }}"
+    - onchanges:
+      - file: {{ test_config }}
```

Comment:
Why not point this `file` entry to the resource name instead of the variable you set? Ie.
```yaml
    - onchanges:
      - file: sshd_config_test_file
```